ogr/services/forgejo/issue.py

```diff
@@ -1,10 +1,167 @@
+        body: str,
+        private: Optional[bool] = None,
+        labels: Optional[list[str]] = None,
+        assignees: Optional[list[str]] = None,
+    ) -> "Issue":
+
+        if private:
+            raise NotImplementedError()
+        if not project.has_issues:
+            raise IssueTrackerDisabled()
+
+        issue = project.service.api.issue.create_issue(
+            owner=project.namespace,
+            repo=project.repo,
+            title=title,
+            body=body,
+            labels=labels,
+            assignees=assignees,
+        )
+        return ForgejoIssue(issue, project)
+
+    @staticmethod
+    def get(project: "forgejo.ForgejoProject", issue_id: int) -> _issue:
+        if not project.has_issues:
+            raise IssueTrackerDisabled
+
+        try:
+            issue = project.service.api.issue.get_issue(
+                owner=project.namespace, repo=project.repo, index=issue_id,
+            )
+        except Exception as ex:
+            raise OperationNotSupported(f"Issue {issue_id} not found") from ex
+        return ForgejoIssue(issue, project)
+
+    @staticmethod
+    def get_list(
+        project: "forgejo.ForgejoProject",
+        status: IssueStatus = IssueStatus.open,
+        author: Optional[str] = None,
+        assignee: Optional[str] = None,
+        labels: Optional[list[str]] = None,
+    ) -> list["Issue"]:
+        if not project.has_issues:
+            raise IssueTrackerDisabled
+
+        parameters: dict[str, Union[str, list[str], bool]] = {
+            "state": status if status != IssueStatus.open else "open",
+            "type": "issues",
+        }
+        if author:
+            parameters["created_by"] = author
+        if assignee:
+            parameters["assigned_by"] = assignee
+        if labels:
+            parameters["labels"] = labels
+
+        issues = project.service.api.issue.list_issues(
+            owner=project.namespace, repo=project.repo, **parameters,
+        )
+        return [ForgejoIssue(issue, project) for issue in issues]
+
+    def close(self) -> "Issue":
+        self._issue_api_call(
+            self.project.service.api.issue.edit_issue,
+            state="closed",
+        )
+        return self
+
+    def add_label(self, *labels: str) -> None:
+        self._issue_api_call(
+            self.project.service.api.issue.add_label,
+            labels=list(labels),
+        )
+
+    def comment(self, body: str) -> IssueComment:
+        comment = self._issue_api_call(
+            self.project.service.api.issue.create_comment,
+            body=body,
+        )
+        return ForgejoIssueComment(self, comment)
+
+    def _get_all_comments(self) -> list[IssueComment]:
+        comments = self._issue_api_call(self.project.service.api.issue.get_comments)
+        return [
+            ForgejoIssueComment(raw_comment=comment, parent=self)
+            for comment in comments
+        ]
+
+    def _issue_api_call(self, method, **kwargs):
```

Comment:
Eh… OK… I would probably even go with something like:

```suggestion
    def _api_call(self, method, **kwargs):
```

Reply:
I’m still new to this and learning the ropes, so I appreciate your patience. I’ll implement the suggested changes and do my best to improve, though there might still be a few rookie mistakes along the way. Thanks for understanding!